x-pack/legacy/plugins/maps/public/layers/sources/es_source.js

```diff
@@ -347,13 +347,16 @@ export class AbstractESSource extends AbstractVectorSource {
   }
 
   getValueSuggestions = async (fieldName, query) => {
-    if (!fieldName) {
+    // fieldName could be an aggregation so it needs to be unpacked to expose raw field.
+    const metricField = this.getMetricFields().find(field => field.getName() === fieldName);
```

Comment:
Not particularly due to this PR, but highlights an older problem imo.

The use of `getMetricFields` seems to be at the wrong the place. It doesn't belong on `AbstractVectorSource`, since only `AbstractESAggSource` can have `AggMetrics`. Up until this PR, `getMetricFields` was never used outside the context of `AbstractESAggSource`.

I would:
- remove the dummy `getMetricFields` method from `AbstractESSource`.
- add new method `getRealFieldName` to `AbstractESSource` and override it in `AbstractESAggSource`. `AbstactESSource#getRealFieldName` would just return `fieldName`. `AbstractESAggSource#getRealFieldBName` does the lookup just like you do here.

Reply:
This change is fixing a bug in getValueSuggestions for aggregated fields. I would like to work on a fix outside of the scope of this PR since this PR is already so large.